### ISSUE TITLE
Clementine shortcut keys triggering when window not in focus

### ISSUE THREAD

**Issue by sueridgepipe** [clementine-player/Clementine#5543]
Clementine 1.3.1-4
Manjaro KDE Linux

When Clementine is running keyboard shortcut keys are always being triggered when Clementine is not in focus.  It doesn't matter if Clementine is minimized, moved to another KDE desktop, or is open in the background, when I type in another application shortcut keys are being triggered.

For instance, N is configured for next song, P for previous song, SPACE for start/stop, etc.  If Clementine is running I cannot type in any other application without these being triggered (and others).

Is this an environment thing, or a Clementine thing.  No other apps do this on my system, to be honest I've not see an app do this on any platform before.

Anyone seen this before?

**Comment by lumpiluk** [comment]
Yes, I've seen this before and I wouldn't want it the other way (except if there was an option to switch between the two behaviors). On my laptop keyboard there are no special keys for next song etc. so I always configure clementine shortcuts to do the job. I haven't experienced any significant collisions with Alt+J, Alt+K, and Alt+L yet.

**Comment by 0Hughman0** [comment]
I've encountered this too, and I don't really like it. A checkbox to stop it from listening when outside of focus would be ideal for me.

**Comment by trevorparsons** [comment]
I've just come back to try Clementine again, and encountered this issue straight away again. 

My keyboard doesn't have dedicated media keys and, being a mouse-reducer, I'd like to be able to set up simple keys to control play/pause (space), previous (page up) and next (page down). But, as sueridgepipe indicates, it very soon becomes obvious that this is impractical. So it's a choice of fiddly mouse control or fiddly custom key combinations, neither of which are ideal.

Obviously whatever happens one will always want the dedicated media keys to be recognised globally.

Amarok seems to deal with this by locating the dedicated media key controls in KDE Global Shortcuts, while intuitive (and practically universal) shortcuts such as hitting the space bar for play/pause are located in "Amarok Shortcuts" and are only effective when the program is focused.

Amarok's [documentation on shortcuts](https://docs.kde.org/trunk5/en/extragear-multimedia/amarok/control-from-console.html#kde-global-shortcuts) isn't quite as explicitly informative on the difference between Global Shortcuts and Amarok Shortcuts as one might like, but it does have a table of the defaults in each.



See
https://docs.kde.org/trunk5/en/extragear-multimedia/amarok/control-from-console.html#kde-global-shortcuts

**Comment by warepilot** [comment]
### Description
Clementine seems to interfiere with other programs when is not focused because it seems to read keystrokes that should recieve only when it's focused.
### Reproduce
For example, if Clementine and LibreOffice are running, LibreOffice won't apply keyboard shortcuts that match those used in Clementine.
(1) Keyboard shortcuts must be changed in one of those programs in order to work in one of them.
(2) If Clementine is closed, LibreOffice shortcuts work properly again. Not happening when other programs run concurrently with LibreOffice and both use same keyshortcuts.
### Versions
Mint Tessa, Clementine 1.3.1

**Comment by melsophos** [comment]
I find this behavior quite painful: I think that shortcuts should never be global, except for few of them and if the user says it explicitly (or if there is a dedicated key for it, like media keys). Otherwise, they should stay within the application.
This is particularly to avoid clashing with other apps which may define a lot of shortcuts (I have dozen of shortcuts when programming for example with Texstudio or Atom).

In principle, I would like to use left / right arrows to more backward / forward in song, but it's clearly not possible with the current setting. And any simple shortcut with arrows will likely collide with another program.

**Comment by leoncryp** [comment]
Global vs Local (in focus) shortcuts

Please create more options for shortcuts, 
1)
eg in VLC has Hotkey and Global, Hotkey only works when app is in focus, Global works anywhere.
Seems Clementine is current hard coded to Global, 
be good to expand more options like VLC ie have Global and Local shortcuts

2) 
Also be good if you can more then 1 shortcuts, eg in SMPlayer, you can assign multiple shortcuts, gives users ability to do  awesome things like mouse gesture control, hot corners etc... run via custom scripts ie xdotool
